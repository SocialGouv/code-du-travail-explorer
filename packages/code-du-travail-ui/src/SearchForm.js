import React from "react";
import PropTypes from "prop-types";

const SearchForm = ({ placeholder, value, onSubmit }) => {
  let input;
  const _onSubmit = e => {
    e.preventDefault();
    onSubmit(input.value);
  };
  const onKeyDown = event => {
    const key = event.key || event.keyCode;
    if (key === "Enter" || key === "13") {
      _onSubmit(event);
    }
  };
  return (
    <form className="search__form">
      <input
        ref={node => (input = node)}
        type="search"
        name="search"
        onKeyDown={onKeyDown}
        defaultValue={value}
        placeholder={placeholder}
        className="search__input"
      />
      <button
        onClick={_onSubmit}
        type="submit"
        className="btn btn__img btn__img__search"
      >
        <span className="hidden">Rechercher</span>
      </button>
    </form>
  );
};

SearchForm.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onSubmit: PropTypes.func
};

SearchForm.defaultProps = {
  placeholder: "Posez votre question",
  value: "",
  onSubmit: () => {}
};

export default SearchForm;
