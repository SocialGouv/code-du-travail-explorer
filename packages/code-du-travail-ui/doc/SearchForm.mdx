---

---

import { Playground, PropsTable } from "docz";
import { SearchForm } from "../src";

## \<SearchForm/\>

<Playground>
  <SearchForm placeholder="Comment puis-je vous aider ?" onSubmit={query => alert(query)}/>
</Playground>

<PropsTable of={SearchForm} />
