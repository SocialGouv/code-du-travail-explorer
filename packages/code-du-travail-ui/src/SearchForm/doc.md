---
name: SearchForm
menu: Component
---
import "@cdt/css";
import { Playground, Props } from "docz";
import SearchForm from ".";

# SearchForm

<Playground>
  <SearchForm
    placeholder="Comment puis-je vous aider ?"
    onSubmit={query => alert(query)}
  />
</Playground>

<Props of={SearchForm} />
