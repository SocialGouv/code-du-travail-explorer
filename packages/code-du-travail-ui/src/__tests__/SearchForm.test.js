import React from "react";
import { render, fireEvent } from "react-testing-library";
import SearchForm from "../SearchForm";

describe("<SearchForm />", () => {
  test("should render", () => {
    const { container } = render(
      <SearchForm
        placeholder="Poser votre question"
        value="démission"
        onSubmit={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("should call onSubmit", () => {
    const handler = jest.fn();
    const { getByPlaceholderText } = render(
      <SearchForm
        placeholder="Poser votre question"
        value="démission"
        onSubmit={handler}
      />
    );
    const input = getByPlaceholderText(/poser votre question/i);
    input.focus();
    fireEvent.change(input, { target: { value: "hello" } });
    fireEvent.keyDown(input, { key: "Enter", code: "Enter" });
    expect(handler).toBeCalledWith("hello");
  });
});
