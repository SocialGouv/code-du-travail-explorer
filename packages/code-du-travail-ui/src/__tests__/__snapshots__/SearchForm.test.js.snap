// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should render 1`] = `
<div>
  <form
    class="search__form"
  >
    <input
      class="search__input"
      name="search"
      placeholder="Poser votre question"
      type="search"
      value="dÃ©mission"
    />
    <button
      class="btn btn__img btn__img__search"
      type="submit"
    >
      <span
        class="hidden"
      >
        Rechercher
      </span>
    </button>
  </form>
</div>
`;
